parser.y

```diff
@@ -80,8 +75,13 @@ entry: main_func {
   ;
 
   /* TODO: mix declarations and statements in compound statement */
-main_func: INT MAIN '(' ')' '{' decls stmts '}' {
-    $$ = std::make_unique<BlockStmtNode>($6, $7);
+main_func: INT MAIN '(' ')' '{' block '}' {
+    $$ = $6;
+  }
+  ;
+
+block: decls stmts {
```

Comment:
```suggestion
block: '{' decls stmts '}' {
```
In the C standard ISO/IEC 9899:1999, section 6.8.2 defines the syntax of a _compound statement_, also known as a _block_.

- $\text{compound-statement}$:
    **{** $\text{block-item-list}_\text{opt}$ **}**
- $\text{block-item-list}$:
    $\text{block-item}$
    $\text{block-item-list}$ $\text{block-item}$
- $\text{block-item}$:
    $\text{declaration}$
    $\text{statement}$

It's important to note that we are combining these 3 non-terminals together.
The key point is that the curly braces (`{`, `}`) are an integral part of the _block_.
This also simplifies references to the _block_ by eliminating the need to include the curly braces every time.
The _block_ always occurs with the curly braces.